GCS backend (#3)
redhat-cop/cloud.terraform_ops: roles/gcs_backend/README.md

# gcs_backend

A role to ensure that the necessary GCP infrastructure is present for an Google Cloud Storage(GCS) remote backend for Terraform.

This role can be used to create or delete following
- A Google Cloud Storage bucket with object versioning enabled

## Role Variables

> - operation - Choices include 'create' and 'delete'. `required`

Reviewer: ```suggestion
- gcs_backend_operation - Choices include 'create' and 'delete'. `required`
```

- gcs_backend_gcp_project - The ID of the Google Cloud Platform project to use. `required`

- gcs_backend_gcs_backend_bucket_name - The name of the bucket to be created. `required`

> - gcs_backend_auth_kind - The type of credential used. Currently supports "serviceaccount". `required`

Reviewer: Why limiting support to service account ?

> - gcs_backend_service_account_file - The path of a Service Account JSON file if serviceaccount is selected as type. `required`

Reviewer: You should add a `gcs_backend_service_account_email` also, this is required when the `auth_kind` is set to `MachineAccount`

# gcs_backend

> A role to ensure that the necessary GCP infrastructure is present for an Google Cloud Storage(GCS) remote backend for Terraform.

Reviewer 2: ```suggestion
A role to ensure that the necessary GCP infrastructure is present for the Google Cloud Storage(GCS) remote backend for Terraform.
```

This role can be used to create or delete following
- A Google Cloud Storage bucket with object versioning enabled

> ## Role Variables

Reviewer 2: Can you also specify the data type of each variable?

> - gcs_backend_operation - Choices include 'create' and 'delete'. Note that `delete` operation might fail if bucket is non-empty. `required`

Reviewer 2: Should we have 'create' as the default value?

# gcs_backend

A role to ensure that the necessary GCP infrastructure is present for the Google Cloud Storage(GCS) remote backend for Terraform.

This role can be used to create or delete following
- A Google Cloud Storage bucket with object versioning enabled

## Role Variables

> 

Reviewer 2: I believe it would be more cohesive to adhere to the format used by the documentation of azure_backend and aws_backend roles-https://github.com/redhat-cop/cloud.terraform_ops/tree/main/roles/aws_s3backend#role-variables.

> - gcs_backend_operation (str) - Choices include 'create' and 'delete'. Note that `delete` operation might fail if bucket is non-empty. `required`

Reviewer 2: default: create

- gcs_backend_service_account_file (str) - The path of a Service Account JSON file if serviceaccount is selected as type. `required` when `gcs_backend_auth_kind = serviceaccount`

- gcs_backend_service_account_email (str) - The service account email used for authentication. `required` when `gcs_backend_auth_kind = machineaccount`.

Return Value
------------

## Examples
```
- name: Create a gcs bucket for Terraform remote backend
  ansible.builtin.include_role:
    name: cloud.terraform_ops.gcs_backend
  vars:
    gcs_backend_operation: create
    gcs_backend_gcs_backend_bucket_name: test-tf-backend-bucket
    gcs_backend_gcp_project: project-uscentral-demo
    gcs_backend_auth_kind: serviceaccount
    gcs_backend_service_account_file: /path/to/auth/credentials.json

- name: Delete the gcs bucket for Terraform remote backend
  ansible.builtin.include_role:
    name: cloud.terraform_ops.gcs_backend
  vars:
    gcs_backend_operation: delete
    gcs_backend_gcs_backend_bucket_name: test-tf-backend-bucket
    gcs_backend_gcp_project: project-uscentral-demo
    gcs_backend_auth_kind: serviceaccount
    gcs_backend_service_account_file: /path/to/auth/credentials.json

> ```

Reviewer 2: Please add the license and author information.

# gcs_backend

> A role to ensure that the necessary GCP infrastructure is present for the Google Cloud Storage(GCS) remote backend for Terraform.

Author: fixed.

This role can be used to create or delete the following
- A Google Cloud Storage bucket with object versioning enabled

## Role Variables

Name | Description | Type | Default | Required
--- | --- | --- | --- | ---

> gcs_backend_operation|Whether to create or delete the Backend resources (GCS bucket). Choices: 'create', 'delete'.  Note that `delete` operation might fail if bucket is non-empty.|string|create| N/A

Author: yes, it is mentioned already 
<img width="1008" alt="image" src="https://github.com/redhat-cop/cloud.terraform_ops/assets/29326601/e841605b-1d3b-45a1-8871-135c3c11c15f">

gcs_backend_gcp_project|The ID of the Google Cloud Platform project to use.|string|N/A| Yes
gcs_backend_bucket_name|The name of the bucket to be created. |string|N/A| Yes

> gcs_backend_auth_kind|The type of credential used|string|N/A| Yes

Reviewer: We should list supported values here (application, machineaccount, serviceaccount)

Author: Fixed

> gcs_backend_service_account_file|The path of a Service Account JSON file if serviceaccount is selected as type.|string|N/A| when `gcs_backend_auth_kind = serviceaccount`

Reviewer: ```suggestion
gcs_backend_service_account_file|The path of a Service Account JSON file if serviceaccount is selected as authentication type.|string|N/A| when `gcs_backend_auth_kind = serviceaccount`
```

Author: Fixed

> gcs_backend_service_account_email|The service account email used for authentication.|string|N/A| when `gcs_backend_auth_kind = machineaccount`

Reviewer: ```suggestion
gcs_backend_service_account_email|An optional service account email address if machineaccount is selected and the user does not wish to use the default email.|string|N/A| No
```

This parameter is not required, even when the auth_kind is set to machineaccount, it offers extra options

Author: Fixed
